drivers/sensor/tmag5273/tmag5273.c

```diff
@@ -0,0 +1,1158 @@
+/*
+ * Copyright (c) 2023 deveritec GmbH
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#define DT_DRV_COMPAT ti_tmag5273
+
+#include "tmag5273.h"
+
+#include <stdint.h>
+#include <stdlib.h>
+
+#include <zephyr/drivers/sensor/tmag5273.h>
+#include <zephyr/dt-bindings/sensor/tmag5273.h>
+
+#include <zephyr/device.h>
+#include <zephyr/kernel.h>
+
+#include <zephyr/drivers/gpio.h>
+#include <zephyr/drivers/i2c.h>
+#include <zephyr/drivers/sensor.h>
+#include <zephyr/sys/byteorder.h>
+#include <zephyr/sys/check.h>
+#include <zephyr/sys/crc.h>
+#include <zephyr/sys/util.h>
+
+#include <zephyr/logging/log.h>
+LOG_MODULE_REGISTER(TMAG5273, CONFIG_SENSOR_LOG_LEVEL);
+
+#define CONV_FACTOR_MT_TO_GS 10
+
+#define TMAG5273_I2C_BUFFER_SIZE                                                                   \
+	(TMAG5273_REG_RESULT_END - TMAG5273_REG_RESULT_BEGIN + 1 +                                 \
+	 COND_CODE_1(CONFIG_CRC, (1), (0)))
```

Comment:
how about:

- having a separate define to 1 to give a name to that `+ 1`
- have a second separate define for the CRC byte size,and then you can set it to 0 or  1 with an `#ifdef CONFIG_CRC`, saves from using `COND_CODE_1`, I think it'd be a bit less obscure

Reply:
If you have a good name for a macro for the `+1` I am open for it.
I have no epiphany for a good name, since this is necessary to get the corrent buffer size. :)

Reply:
what's that byte for? status bits?

Reply:
`TMAG5273_REG_RESULT_BYTE_SIZE`? and then if there's a define for it you could add a comment explaining it next to it

Reply:
I've decided to just add a better documentation instead of using another `define` which I would need to document anyway.